src/lib/cache/util.go

```diff
@@ -57,3 +60,48 @@ func (km keyMutex) Unlock(key interface{}) {
 	km.m.Delete(key)
 	m.Unlock()
 }
+
+// ObjectKey normalizes cache object key.
+type ObjectKey struct {
+	// namespace as group or prefix, eg. artifact:id
+	namespace string
+}
+
+// NewObjectKey returns object key with namespace.
+func NewObjectKey(ns string) *ObjectKey {
+	return &ObjectKey{namespace: ns}
+}
+
+// Format formats fields to string.
+// eg. namespace: 'artifact'
+// Format("id", 100, "digest", "aaa"): "artifact:id:100:digest:aaa"
+func (ok *ObjectKey) Format(keysAndValues ...interface{}) (string, error) {
+	// keysAndValues must be paired.
+	if len(keysAndValues)%2 != 0 {
+		return "", errors.Errorf("invalid keysAndValues: %v", keysAndValues...)
+	}
+
+	s := ok.namespace
+	for i := 0; i < len(keysAndValues); i++ {
+		// even is key
+		if i%2 == 0 {
+			key, ok := keysAndValues[i].(string)
```

Comment:
Rename ok to another name to avoid conflict

Reply:
good suggestion